libraries/AP_NavEKF2/AP_NavEKF2_PosVelFusion.cpp

```diff
@@ -602,10 +602,15 @@ void NavEKF2_core::FuseVelPosNED()
             varInnovVelPos[5] = P[8][8] + R_OBS_DATA_CHECKS[5];
             // calculate the innovation consistency test ratio
             hgtTestRatio = sq(innovVelPos[5]) / (sq(MAX(0.01f * (float)frontend->_hgtInnovGate, 1.0f)) * varInnovVelPos[5]);
-            // fail if the ratio is > 1, but don't fail if bad IMU data
-            hgtHealth = ((hgtTestRatio < 1.0f) || badIMUdata);
+
+            // when on the ground we accept a larger test ratio
```

Comment:
Would be worth adding a few words to explain why we are doing this, eg

// when on ground we accept a larger test ratio to allow the filter to handle 
// large switch on IMU bias errors without rejecting the height sensor

Reply:
will do. I didn't actually know why it was letting them through unchecked. Now I know I can add a comment :-)